### ISSUE TITLE
LLaVA-NeXT fine-tuning collator does not mask image tokens from the loss

### ISSUE THREAD

**Issue by Mayukhga83** [NielsRogge/Transformers-Tutorials#511]
Hi, thanks for providing these tutorials.

I noticed a potential issue in:

LLaVa-NeXT/Fine_tune_LLaVaNeXT_on_a_custom_dataset_(with_PyTorch_Lightning).ipynb

The training collator currently creates labels as follows:

labels = batch["input_ids"].clone()
labels[labels == processor.tokenizer.pad_token_id] = -100
batch["labels"] = labels

This masks padding tokens, but leaves the LLaVA image-token positions in the labels.

With recent Transformers versions, the processor expands the <image> placeholder into image-token positions before the inputs reach the model. As a result, these positions can also contribute to the causal language-modeling loss when labels is simply copied from input_ids.

Current Hugging Face VLM fine-tuning examples additionally mask the image token ID from the labels.

Would it therefore make sense to change the collator to something similar to:

labels = batch["input_ids"].clone()

labels[labels == processor.tokenizer.pad_token_id] = -100

image_token_id = processor.tokenizer.convert_tokens_to_ids(
    processor.image_token
)
labels[labels == image_token_id] = -100

batch["labels"] = labels

Alternatively, model.config.image_token_index could be used if that is preferable for LLaVA-NeXT.

This seems especially relevant now that the repository environment specifies transformers>=4.57.3, where multimodal token expansion is handled by the processor.

Without masking these positions, potentially many visual placeholder tokens per image participate in the text-generation loss even though they are conditioning positions rather than desired textual targets.

Thanks!

**Comment by NielsRogge** [comment]
Thanks for the careful report — this is correct.

With recent Transformers versions the processor expands the `<image>` placeholder into many image-token positions, so copying `input_ids` into `labels` and only masking padding lets those visual tokens contribute to the LM loss.

I updated the LLaVA-NeXT training collator to also mask the image token ID with `-100`, following the pattern you suggested:

```python
labels = batch["input_ids"].clone()
labels[labels == processor.tokenizer.pad_token_id] = -100
image_token_id = processor.tokenizer.convert_tokens_to_ids(processor.image_token)
labels[labels == image_token_id] = -100
batch["labels"] = labels
```

Fix is in #513.
